src/navbar/types.ts

```diff
@@ -4,10 +4,19 @@ import { TextLinkProps } from "../text";
 // =============================================================================
 // NAV ITEM TYPES
 // =============================================================================
-export interface NavItemProps<T> extends NavItemCommonProps<T> {
+
+export interface NavItemLinkProps<T> extends NavItemCommonProps<T> {
+    itemType?: "link";
     subMenu?: NavItemCommonProps<T>[] | undefined;
 }
 
+export interface NavItemComponentProps {
+    itemType: "component";
+    children: JSX.Element;
```

Comment:
actually `children` in the `TextLinkProps` allows for JSX elements too. I would go with using `NavItemLinkProps` as a single interface with an optional prop called `renderAs` where it only allows `"link" | "custom-component"` and it defaults to `"link"`